src/backend/distributed/README.md

```diff
@@ -1955,12 +1955,153 @@ The distributed transactionId and backend/PID mapping is done via BackendData st
+
+#### Citus Advisory Locks 3: Query from any node
+  
+When users are allowed to run queries from any node, then in certain cases, we need to form a synchronization that involes multiple nodes. Advisory locks is a convinent tool for achieving these types of goals.
+
+ Citus exposes few UDFs like `lock_shard_resources()` and `lock_shard_metadata()` which are simple wrappers around the metadata and executor locks we discussed above. 
+
+ When there are nodes with metadata, then Citus acquires some of the advisory locks on all nodes, like:
+
+ ```sql
+  select citus_move_shard_placement(102008, 'localhost', 9701, 'localhost', 9702, shard_transfer_mode:='force_logical');
+  ....
+
+  DETAIL:  on server onderkalaci@localhost:9702 connectionId: 2
+NOTICE:  issuing SELECT lock_shard_metadata(7, ARRAY[102008])
+DETAIL:  on server onderkalaci@localhost:9701 connectionId: 1
+NOTICE:  issuing SELECT lock_shard_metadata(7, ARRAY[102008])
+DETAIL:  on server onderkalaci@localhost:9702 connectionId: 2
+...
+```
+
+Another useful application of advisory locks in query from any node is modifications to reference tables. Reference tables (or in general replicated tables) have multiple placements for a given shard. So, modifications to the placements of the same shard should be serialized. We cannot allow multiple modification commands to modify the placements in different orders. It could cause diverging the contents of the data.
+
+The coordinator already serializes the modifications to reference tables via `LockShardResource()` C function. When there are other nodes in the cluster, Citus sends the similar command, to the first worker node in the cluster. In general, Citus aims to serialize operations on the reference tables via acquiring advisory locks on the first worker node, see `SerializeNonCommutativeWrites()` and `LockShardListResourcesOnFirstWorker()` C functions. The outcome is the following:
+
+```sql
+ BEGIN;
+BEGIN
+-- 9701 is the first worker node in the metadata
+insert into reference_table VALUES (1);
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 8, '2023-09-18 16:30:18.715259+03');
+DETAIL:  on server onderkalaci@localhost:9701 connectionId: 1
+NOTICE:  issuing SELECT lock_shard_resources(3, ARRAY[102040])
+DETAIL:  on server onderkalaci@localhost:9701 connectionId: 1
+NOTICE:  issuing INSERT INTO public.reference_table_102040 (a) VALUES (1)
+
+``` 
+
```

Comment:
maybe add a bit more context / critique of the first node issue, as well as writes to replicated tables